### PR TITLE
Remove BindingAnnotation in favor of SqlStatementCustomizingAnnotation

### PR DESCRIPTION
**Pull request by qualidafial**
Hello, techies.

Look at your binders, now back to customizers. Now back at your binders, now back to customizers.

Sadly, your binders aren't customizers, but if you stopped using @BindingAnnotation and switched to
@SqlStatementCustomizingAnnotation, they could _smell_ like customizers.

Look down, back up, where are you?

You're on a code base with the customizers your binders could smell like.

What's in your Github notifications, back at me. I have it, it's a pull request to that refactor you love.

Look again, the binders are now customizers!

Anything is possible when your binders smell like customizers and not a redundant design feature.

I'm on a chair.

### CODE REVIEW

**Comment by qualidafial** [comment]
https://youtu.be/VX5au0LOJp8

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/BatchHandler.java [hunk 1]
```diff
@@ -142,9 +142,8 @@ public boolean hasNext() {
                 }
                 // execute a single chunk and buffer
                 PreparedBatch batch = handle.prepareBatch(sql);
-                applyCustomizers(batch, args);
                 for (int i = 0; i < chunkSize && batchArgs.hasNext(); i++) {
-                    applyBinders(batch.add(), batchArgs.next());
+                    applyCustomizers(batch.add(), batchArgs.next());
```
Comment (qualidafial) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/BatchHandler.java]: Note that this changes means that _all_ method parameters are now batch-able to a `@SqlBatch` method--so users may have to tack on a `@SingleValue` for things like `@BindList` or `@DefineList`.

**Comment by qualidafial** [comment]
@stevenschlansker @arteam This is ready for review

File: jpa/src/main/java/org/jdbi/v3/jpa/BindJpa.java [hunk 2]
```diff
@@ -13,18 +13,66 @@
...
+                                                         Method method,
+                                                         Parameter param,
+                                                         int index,
+                                                         Object arg) {
+            BindJpa bind = (BindJpa) annotation;
+            return stmt -> {
+                final String prefix;
+                if (bind.value().isEmpty()) {
+                    prefix = "";
+                } else {
```
Comment (stevenschlansker) [inline jpa/src/main/java/org/jdbi/v3/jpa/BindJpa.java]: this block should probably be lifted out of the lambda?

**Pull request merged**